<launch>

    <!-- launch turtle_pent.launch file -->
    <include file="$(find tsim)/launch/turtle_pent.launch">
    </include>

    <!-- load the paramters, wheel_base, wheel_radius, etc. -->
    <rosparam command="load" file="$(find nuturtle_description)/config/diff_params.yaml"/>

    <!--start the turtlesim and turtle_rect nodes-->
    <node pkg="rigid2d" name="odometer" type="odometer" output="screen"/>
    <node pkg="rigid2d" name="fake_diff_encoders" type="fake_diff_encoders" output="screen"/>

    <!--robot description-->
    <param name="robot_description" command="$(find xacro)/xacro --inorder $(find nuturtle_description)/urdf/diff_drive.urdf.xacro"/>  
    
    <!-- node to publish the robot state -->
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>
    
    <!--nodes from rviz with rviz configurations-->
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find nuturtle_description)/config/turtle_odom_pent.rviz" required="true"/>

</launch>
